UnitedIncome/components
# [APP-2985] Removes the requirement for react-router-dom

### Description
> Please provide a brief description of your changes below.

This pull request removes the `react-router-dom` dependency from the component library. We've run into problems several times now where we're blocking releases of this package due to a mismatch in version numbers between our client and component library, they essentially have to always be updated at the same time which isn't ideal.

The react-router-dom package gets used in the `Button` component when a link is used. What I've instead done is create a `linkComponent` prop which can be passed into the Button component, and if it doesn't exist a basic anchor tag is rendered in its place.

In the client I've created a wrapper for the Button which will automatically pass in the react-router-dom `Link` component whenever it's referenced. Utilizing this change will allow us (and anyone else) who doesn't utilize react-router-dom in their project to still use the buttons.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

This is more easily tested in the client, but you can try it in Storybook too. There is a seperate PR for the client changes.

### Comments
> Any additional comments that will aid with the review process.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-2985)
